hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java

```diff
@@ -392,6 +397,24 @@ private static void initializeUsedBytesAndBlockCount(DatanodeStore store,
     kvData.setBlockCount(blockCount);
   }
 
+  /**
+   * A container is empty if:
+   * - The container is closed
+   * - There are no blocks in its block table.
+   *
+   * Empty containers are eligible for deletion.
+   */
+  public static boolean isEmpty(DatanodeStore store,
+      KeyValueContainerData container) throws IOException {
+    if (container.isOpen()) {
+      return false;
+    }
+    try (BlockIterator<BlockData> blockIterator =
```

Comment:
This PR is being done and added a flag due to inconsistency in block count data, so what brings inconsistency and why that inconsistency will not impact above logic ?

Reply:
We keep block count metadata information separately about container, which can go wrong sometime due to various reasons. Now we decouple metadata information to judge if container is empty or not by checking directly block table if any block exists or not for the particular container.